RevenueCatUI/UIKit/PaywallViewController.swift

```diff
@@ -196,4 +202,54 @@ public protocol PaywallViewControllerDelegate: AnyObject {
 
 }
 
+// MARK: - Private
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+private extension PaywallViewController {
+
+    func createHostingController() -> UIHostingController<PaywallContainerView> {
+        let controller = UIHostingController(rootView: PaywallContainerView(configuration: self.configuration,
+                                                                            controller: self))
+
+        // make the background of the container clear so that if there are cutouts, they don't get
+        // overridden by the hostingController's view's background.
+        controller.view.backgroundColor = .clear
+        controller.view.translatesAutoresizingMaskIntoConstraints = false
+
+        return controller
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+private struct PaywallContainerView: View {
+
+    var configuration: PaywallViewConfiguration
+    var controller: PaywallViewController
```

Comment:
TODO: need to test to make sure this doesn't lead to a leak due to this retain cycle.

Reply:
Tested with Cesar and we fixed it.